tests/payload/test_manifest.py

```diff
@@ -0,0 +1,306 @@
+
+
+def test_manifest_parse_obj(manifest_obj, manifest_dict):
+    assert Manifest.parse_obj(manifest_dict) == manifest_obj
+
+
+@mock.patch(
+    "yapapi.payload.manifest.datetime", **{"utcnow.return_value": datetime(2020, 1, 1, tzinfo=UTC)}
+)
+def test_manifest_with_minimal_data(mocked_datetime):
+    payload_hash = "asd"
+
+    minimal_manifest_dict = Manifest(
+        payload=[
+            ManifestPayload(
+                hash=payload_hash,
+            ),
+        ],
+    ).dict()
+
+    assert minimal_manifest_dict == {
+        "created_at": "2020-01-01T00:00:00+00:00",
+        "expires_at": "2100-01-01T00:00:00+00:00",
+        "payload": [
+            {
+                "hash": payload_hash,
+                "urls": [],
+                "platform": {
+                    "arch": "x86_64",
+                    "os": "linux",
+                },
+            },
+        ],
+        "version": "",
+    }
+
+
+@mock.patch(
+    "yapapi.payload.manifest.datetime", **{"utcnow.return_value": datetime(2020, 1, 1, tzinfo=UTC)}
+)
+def test_manifest_generate(mocked_datetime):
+    payload_urls = [
+        "some url",
+        "some other url",
+        "yet another url",
+    ]
+    payload_hash = "some hash"
+    metadata_name = "example"
+    metadata_version = "0.0.1"
+    comp_manifest_script_match = "strict"
+
+    assert Manifest.parse_imploded_obj(
+        {
+            "metadata.name": metadata_name,
+            "metadata.version": metadata_version,
+            "payload.0.hash": payload_hash,
+            "payload.0.urls": [
+                payload_urls[0],
+                payload_urls[1],
+            ],
+            "payload.0.urls.2": payload_urls[2],
+            "comp_manifest.script.match": comp_manifest_script_match,
+        }
+    ).dict() == {
+        "metadata": {
+            "name": metadata_name,
+            "version": metadata_version,
+            "authors": [],
+            "homepage": None,
+            "description": None,
+        },
+        "comp_manifest": {
+            "script": {
+                "commands": [
+                    "run .*",
+                ],
+                "match": comp_manifest_script_match,
+            },
+            "version": "",
+        },
+        "created_at": "2020-01-01T00:00:00+00:00",
+        "expires_at": "2100-01-01T00:00:00+00:00",
+        "payload": [
+            {
+                "hash": payload_hash,
+                "urls": payload_urls,
+                "platform": {
+                    "arch": "x86_64",
+                    "os": "linux",
+                },
+            },
+        ],
+        "version": "",
+    }
+
+
+@pytest.mark.asyncio
+async def test_manifest_payload_resolve_urls_from_hash():
+    payload = ManifestPayload(
+        hash="05270a8a938ff5f5e30b0e61bc983a8c3e286c5cd414a32e1a077657",
+    )
+
+    assert payload.urls == []
+
+    await payload.resolve_urls_from_hash()
```

Comment:
as far as I can see, this _unit_ test pulls data from the actual Golem repository, please mock the appropriate response so that this test doesn't directly depend on an external resource, thank you :)